data/dataloader.py

```diff
@@ -0,0 +1,48 @@
+from type import (
+    PreprocessConfig,
+    DatasetCategories,
+    TrainDataset,
+    TestDataset,
+)
+from typing import List, Callable, Tuple, Optional, Union
+from .transformation import transform_dataset
+from datasets.load import load_dataset
+
+from .merge import merge_datasets
+
+
+class DataLoaderBase:
+    def load(self, category: DatasetCategories) -> Union[TrainDataset, TestDataset]:
+        pass
+
+
+class DataLoader(DataLoaderBase):
+    def __init__(
+        self,
+        path: str,
+        preprocessing_config: PreprocessConfig,
+        name: Optional[str] = None,
+        transformations: List[Callable] = [],
+    ):
+        obligatory_transformations = [transform_dataset]
+
+        self.transformations = obligatory_transformations + transformations
+        self.preprocessing_config = preprocessing_config
+        self.data = load_dataset(path, name)
+
+    def load(self, category: DatasetCategories) -> Union[TrainDataset, TestDataset]:
+        for transformation in self.transformations:
+            self.data = transformation(self.data, self.preprocessing_config)
+        return self.data[category]
+
+
+class DataLoaderMerger(DataLoaderBase):
```

Comment:
maybe we could do this automatically if you pass in a list of DataLoaders? althought it's probabl fine this way